src/traces/image/plot.js

```diff
@@ -139,7 +139,10 @@ module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
         // Pixelated image rendering
         // http://phrogz.net/tmp/canvas_image_zoom.html
         // https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering
-        var style = 'image-rendering: optimizeSpeed; image-rendering: -moz-crisp-edges; image-rendering: -o-crisp-edges; image-rendering: -webkit-optimize-contrast; image-rendering: optimize-contrast; image-rendering: crisp-edges; image-rendering: pixelated;';
+        var style = '';
+        if(trace.interpolate === 'nearest') {
+            style = 'image-rendering: optimizeSpeed; image-rendering: -moz-crisp-edges; image-rendering: -o-crisp-edges; image-rendering: -webkit-optimize-contrast; image-rendering: optimize-contrast; image-rendering: crisp-edges; image-rendering: pixelated;';
```

Comment:
It is little hard to follow this long line.
Let's add
```
pixelatedStyle: [
    'image-rendering: optimizeSpeed',
    'image-rendering: -moz-crisp-edges'
    'image-rendering: -o-crisp-edges'
    'image-rendering: -webkit-optimize-contrast'
    'image-rendering: optimize-contrast'
    'image-rendering: crisp-edges'
    'image-rendering: pixelated'
].join('; ')
```
to constants, then use it here
```js
var style = (trace.zsmooth === false) ? constants.pixelatedStyle : '';
```